### ISSUE TITLE
Training-related issues

### ISSUE THREAD

**Issue by LumaFilter** [Ethan-Tseng/Neural_Nano-Optics#6]
Thanks to the authors for sharing such fantastic work!
1. I've been attempting to replicate your article, and I've followed the configuration outlined in the supplementary materials: PHASE_ITERS=10, G_ITERS=100, METASURFACE=neural. I trained the code on a single RTX 3090 GPU, but I've consistently observed that the loss function fails to decrease, oscillating between 0.7 and 0.8. I'm wondering if I used the correct configuration.
2. By reading through the code, I've noticed that the parameters are updated only once after training PHASE_ITERS or G_ITERS, i.e. batch_size=1. Could it be possible that the loss oscillation is due to the batch size being too small?

**Comment by weixiang-zhang** [comment]
Hi, Is your problem solved? Could you share some of your solution? I am facing the similar issue of convergence. Thanks！

**Comment by vrk7** [comment]
@LumaFilter @weixiang-zhang were you able to run the code?
